port/unix/omrsignal.c

```diff
@@ -699,119 +727,55 @@ asynchSignalReporter(void *userData)
 
 	omrthread_set_name(omrthread_self(), "Signal Reporter");
 
-	for (;;) {
-		Trc_PRT_signal_omrsig_asynchSignalReporterThread_going_to_sleep();
-
-#if defined(J9ZOS390)
-		/* I won't attempt to generate diagnostics if the following pthread
-		 * functions return errors because it may interfere with diagnostics
-		 * we are attempting to generate for earlier events.
-		 */
-		pthread_mutex_lock(&wakeUpASyncReporterMutex);
-		Trc_PRT_signal_omrsig_asynchSignalReporter_woken_up();
-
-		for (;;) {
-			if (0 != signalCounts[SIGQUIT_PENDING]) {
-				Trc_PRT_signal_omrsig_asynchSignalReporter_woken_up_for_SIGQUIT();
-				asyncSignalFlag = OMRPORT_SIG_FLAG_SIGQUIT;
-				signalCounts[SIGQUIT_PENDING]--;
-				break;
-			} else if (0 != signalCounts[SIGABRT_PENDING]) {
-				Trc_PRT_signal_omrsig_asynchSignalReporter_woken_up_for_SIGABRT();
-				asyncSignalFlag = OMRPORT_SIG_FLAG_SIGABRT;
-				signalCounts[SIGABRT_PENDING]--;
-				break;
-			} else if (0 != signalCounts[SIGTERM_PENDING]) {
-				Trc_PRT_signal_omrsig_asynchSignalReporter_woken_up_for_SIGTERM();
-				asyncSignalFlag = OMRPORT_SIG_FLAG_SIGTERM;
-				signalCounts[SIGTERM_PENDING]--;
-				break;
-			} else if (0 != signalCounts[SIGXFSZ_PENDING]) {
-				Trc_PRT_signal_omrsig_asynchSignalReporter_woken_up_for_SIGXFSZ();
-				asyncSignalFlag = OMRPORT_SIG_FLAG_SIGXFSZ;
-				signalCounts[SIGXFSZ_PENDING]--;
-				break;
-			} else if (0 != shutDownASynchReporter) {
-				Trc_PRT_signal_omrsig_asynchSignalReporter_woken_up_for_shutdown();
-				asyncSignalFlag = 0;
-				break;
-			} else {
-				pthread_cond_wait(&wakeUpASyncReporterCond, &wakeUpASyncReporterMutex);
-			}
-		}
-		pthread_mutex_unlock(&wakeUpASyncReporterMutex);
-
-		/* we get woken up if there is a signal pending or it is time to shutdown */
-		if (0 != shutDownASynchReporter) {
-			break;
-		}
-#else /* defined(J9ZOS390) */
+	while (0 == shutDownASynchReporter) {
+		int unixSignal = 1;
+		uint32_t asyncSignalFlag = 0;
 
+#if !defined(J9ZOS390)
```

Comment:
this could just be an #else

Reply:
inside the `while` loop, there is no prior`ifdef J9ZOS390`. do you want me to move the `sem_wait` after the `for` loop along with the `cond_wait`?

Reply:
No. Sorry, I misread the folded code.